react/features/calendar-sync/components/ConferenceNotification.native.js

```diff
@@ -0,0 +1,242 @@
+                            </View>
+                            <View style = { styles.notificationIconContainer }>
+                                <Icon
+                                    name = 'navigate_next'
+                                    style = { styles.notificationIcon } />
+                            </View>
+                        </View>
+                    </TouchableOpacity>
+                </View>
+            );
+        }
+
+        return null;
+    }
+
+    _getContainerStyle: () => Array<Object>;
+
+    /**
+     * Assembles a style array for the container.
+     *
+     * @private
+     * @returns {Array<Object>}
+     */
+    _getContainerStyle() {
+        const { event } = this.state;
+        const now = Date.now();
+        const style = [
+            styles.notificationContainer,
+            {
+                top: this.props._screenHeight / 10
+            }
+        ];
+
+        if (event && event.startDate < now && event.endDate > now) {
+            style.push(styles.notificationContainerPast);
+        }
+
+        return style;
+    }
+
+    _maybeDisplayNotification: () => void;
+
+    /**
+     * Periodically checks if there is an event in the calendar for which we
+     * need to show a notification.
+     *
+     * @private
+     * @returns {void}
+     */
+    _maybeDisplayNotification() {
+        const { _currentConferenceURL, _eventList } = this.props;
+        let eventToShow;
+
+        if (_eventList && _eventList.length) {
+            const now = Date.now();
+
+            for (const event of _eventList) {
+                if (event.url !== _currentConferenceURL) {
+                    if ((!eventToShow
+                        && event.startDate > now
+                        && event.startDate < now + ALERT_MILLISECONDS)
+                        || (event.startDate < now && event.endDate > now)
+                    ) {
+                        eventToShow = event;
+                    }
+                }
+            }
+        }
+
+        this.setState({
+            event: eventToShow
+        });
+    }
+
+    _onGoToNext: () => void;
+
+    /**
+     * Opens the meeting URL that the notification shows.
+     *
+     * @private
+     * @param {string} url - The URL to open.
+     * @returns {void}
+     */
+    _onGoToNext() {
+        const { event } = this.state;
+
+        if (event && event.url) {
+            this.props.dispatch(appNavigate(event.url));
+        }
+    }
+
+}
+
+/**
+ * Maps redux state to component props.
+ *
+ * @param {Object} state - The redux state.
+ * @returns {{
+ *      _eventList: Array,
+ *      _screenHeight: number
+ * }}
+ */
+export function _mapStateToProps(state: Object) {
+    const { height } = Dimensions.get('window');
+    const { locationURL } = state['features/base/connection'];
+
+    return {
+        _currentConferenceURL:
+            locationURL ? getURLWithoutParams(locationURL)._url : '',
```

Comment:
When does it happen that locationURL is falsy? If _currentConferenceURL is an empty string, does the searching done in _maybeDisplayNotification encounter any problems?

Also, is there any guaranteed order the eventList will be in? I'm wondering if it's possible to do an early exit of _maybeDisplayNotification if there a guarantee of time order.

Reply:
Regarding the falsy value, I just assumed it can be null, based on reducer.js:195 in base/connection

About the notification, yes, it was the case in the past, but based on Lyubo's request (see above) I quite changed the order of displaying the notification. See comment "Avoid asking for calendar..."